lib/os_io.h

```diff
@@ -620,7 +629,8 @@ extern ___SCMOBJ ___device_stream_setup
         ());
 
 extern ___SCMOBJ ___device_select
-   ___P((___device **devs,
+   ___P((___processor_state ___ps,
```

Comment:
I thought we agreed to not pass ```___ps``` as a parameter because it is needed in too many places where it is awkward to propagate to.  I prefer all or nothing.